### PR TITLE
added Debug mode to the Module

### PR DESCRIPTION
**Pull request by jac1013**
## Story / Task

https://github.com/bixlabs/bix-notifications/issues/7

## What this PR does?

* It adds Debug mode for the module.

### Before

* We didn't have Debug mode.

### After

* We have Debug mode now.

## How to test / reproduce

* We can't test it yet :(.

## TODOs

* At least see that this is working

Thanks.

### CODE REVIEW

**Comment by jac1013** [comment]
Hey @rkmax I just added you as a reviewer because I would like your explanation/opinion of how this code can be tweak to accept promises and callbacks.

Thanks!

**Comment by rkmax** [comment]
@jac1013 first about the debug module, what I know about this module it was intuited by observing other projects that use it and the way in which the variable DEBUG works. instead having verbosity levels like info, error, warn, etc (things from java xD I think)  they use "namespaces" basically are `:` to explain concerns to print on the console log, and normally all the "debug strings" uses a prefix, normally something like the module name that never changes (I think it isn't necessary to define an ENV variable for that) instead that we debug for parts of the applications, an example

- `module:model:user`
- `module:controller:user`
- `module:controller:notification`

so we can enable logs for specific parts of the app

Print for all controllers

    DEBUG=module:controller:*

Print for all controller and model user

    DEBUG=module:controller:*,module:model:user

second, regarding the promises/callbacks ... [WIP]

File: src/notifications/controller.js [hunk 1]
```diff
@@ -7,73 +9,132 @@ class NotificationController {
...
    * Creates a new notification.
    * @param  {Object} data, object representing a notification.
    */
-  create(data) {
-    return this.db.create(data);
+  async create(data) {
+    try {
+      const notification = await this.db.create(data);
+      Debug.info(`Notification was created at ${new Date()} with values: ${JSON.stringify(notification)}`);
+      return notification;
```
Comment (rkmax) [inline src/notifications/controller.js]: I think you can define the method like

```js
async create(data, callback = null) {
  // ...
  const hasCallback = typeof callback === 'function';
  if (hasCallback) {
    notification
      .then((notification) => callback(null, notification))
      .catch(callback)
   } else {
    return notification
  }
}
```

**Review by rkmax — COMMENTED** [review]
regarding the callback/promise style I've added an example

File: src/notifications/controller.js (hunk 1 above)
Comment (jac1013) [reply src/notifications/controller.js]: Nice, thanks for the example, amazing!

**Comment by jac1013** [comment]
@rkmax I made some changes for the callback/promise thing, I can't test this yet sadly, but with this https://github.com/bixlabs/bix-notifications/pull/20 I would be able to, let's wait until we have that merge, let me know if you see anything weird in the code.

EDIT: I removed the possibility of using an environment variable to change the Debug String that the modules use (just to keep things simpler).

File: src/notifications/util.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+async function tryCallbackOrPromise(promise, callback = null) {
+  if (isValidCallback(callback)) {
+    promise.then((data) => {
+      callback(null, data);
+    })
+      .catch(callback);
+  }
+  await promise;
```
Comment (rkmax) [inline src/notifications/util.js:8]: @jac1013 dont should return it the promise here instead use async/await?

Comment (jac1013) [reply src/notifications/util.js:8]: I will test this later correctly, right now we can't tests this changes, probably you are right, I'll let you know whenever I test this.

**Pull request merged**